File: cmd/site.go
package cmd

import (
	"log"
	"os"
	"os/exec"

	"github.com/spf13/cobra"
)

func openBrowser(url string) {
	var err error
	exec.Command("xdg-open", url).Start()
	if err != nil {
		log.Fatal(err)
	}
	os.Exit(0)
}

var (
	siteCmd = &cobra.Command{
		Use:   "site",
		Short: "Abre os horrorosos sites da UFABC",
		Run: func(cmd *cobra.Command, args []string) {
			if len(args) == 0 {
				cmd.Help()
				os.Exit(-1)
			}

			switch args[0] {
			case "tidia":
				openBrowser("https://tidia4.ufabc.edu.br/portal/relogin")
			case "prograd":
				openBrowser("https://prograd.ufabc.edu.br/")
			case "gradmat":
				openBrowser("https://gradmat.ufabc.edu.br/")
			case "bcc":
				openBrowser("https://bcc.ufabc.edu.br")
			case "sigaa":
				openBrowser("https://sig.ufabc.edu.br/sigaa/verTelaLogin.do")
			case "moodle":
				openBrowser("https://moodle.ufabc.edu.br/")
			case "neuro":
				openBrowser("https://neuro.ufabc.edu.br/")
			case "ufabc":
				openBrowser("https://www.ufabc.edu.br/")
			}

		},
	}
)

func init() {
	rootCmd.AddCommand(siteCmd)
}

File: nyx/site.py
from enum import Enum
import click


class WebsitesUFABC(Enum):

    bcc = 'https://bcc.ufabc.edu.br'
    gradmat = 'https://gradmat.ufabc.edu.br/'
    moodle = 'https://moodle.ufabc.edu.br/'
    neuro = 'https://neuro.ufabc.edu.br/'
    prograd = 'https://prograd.ufabc.edu.br/'
    sigaa = 'https://sig.ufabc.edu.br/sigaa/verTelaLogin.do'
    tidia = 'https://tidia4.ufabc.edu.br/portal/relogin'
    ufabc = 'https://www.ufabc.edu.br/'


def make_website_subcommand(name, url):

    fn_template = f'''@site.command()
def {name}():
    """Abre o site do(a) {name}"""
    click.launch('{url}')
    '''

    exec(fn_template)
    website_subcommand = locals()[f'{name}']

    return website_subcommand


@ click.group('site')
def site():
    """Utilitarios de Sites da UFABC"""
    ...


for website in WebsitesUFABC:
    make_website_subcommand(website.name, website.value)
